test/unit/org/apache/cassandra/utils/RecomputingSupplierTest.java

```diff
@@ -93,31 +89,40 @@ public void recomputingSupplierTest() throws Throwable
                             lastSeen.accumulateAndGet(v, Math::max);
                             Assert.assertTrue(String.format("Seen %d out of order. Last seen value %d", v, seenBeforeGet),
                                               v >= seenBeforeGet);
-
                         }
-
                     }
                     catch (Throwable e)
                     {
                         thrown.set(e);
-                        latch.countDown();
                     }
                 }
             });
         }
 
-        latch.await(10, TimeUnit.SECONDS);
-
+        Util.spinAssertEquals(true, () -> {return counter.get() > 1;}, 11);
```

Comment:
nit: I'd probably take intelij's suggestion and replace the second argument with `() -> counter.get() > 1` which is a little less verbose.

Reply:
Eclipse complains if I don't use the `return` :shrug:

Reply:
Sorry for nitpicking then!

Reply:
I understand that the purpose of this line is ensuring that the tasks above have had time to update `counter`. The previous incarnation of this line awaited for 10s which is probably too much, but I'm wondering if waiting for a single increase of counter is too little. Maybe we can wait until it has been increased more times, for example:
```suggestion
        Util.spinAssertEquals(true, () -> {return counter.get() > 1000;}, 11);
```

Reply:
> nit: I'd probably take intelij's suggestion and replace the second argument with () -> counter.get() > 1 which is a little less verbose.

I'd also go with the concise format, `() -> counter.get() > 1`. I haven't seen any rule about this in our code style, so I guess that the IDE warning depends on random defaults not covered by the project's code style. The `eclipse-warnings` Ant target seems to accept both formats. Anyway, I haven't found anything on the codebase containing the pattern `) -> {return`, while there are numerous usages of the compact format. So, we can probably assume that that's the untold convention.

Reply:
Mmmm I must have made a typo before bc I tested this now again and it doesn't complain :shrug: Changed.

Reply:
On the waiting of 11s it is a left over of a previous approach of 10 threads with a 1s sleep bla bla tes I was doing. So I was waiting 11s to give time to all 10 threads. Makes no sense now and 10 reads better, right.

On the purpose of that line giving 10s of time to recompute I am not sure. Why not just sleep? Why wait for an exception that never happens? I never grasped fully that approach.